coalib/settings/SectionFilling.py

```diff
@@ -45,7 +45,10 @@ def fill_section(section, acquire_settings, log_printer, bears):
 
     # Get missing ones.
     if len(needed_settings) > 0:
-        new_vals = acquire_settings(None, needed_settings, section)
+        if acquire_settings.__name__ == 'fail_acquire_settings':
```

Comment:
this is a bad check ;) Only functions with this name are then called with the old calling strategy.
Best would be to use function inspections and detect how many parameters are accepted, and choose with that information the right "calling convention".

Reply:
@Makman2 only the fail_acquire_settings function accepts 2 parameters . Rest all accept 3